src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -450,35 +451,55 @@ protected JiraSession createSession() {
 
         String userName = credentials.getUsername();
         Secret password = credentials.getPassword();
+
+        final JiraRestClient jiraRestClient = new AsynchronousJiraRestClientFactory()
+                .create(uri, new BasicHttpAuthenticationHandler( userName, password.getPlainText()),getHttpClientOptions());
+        return new JiraSession(this, new JiraRestService(uri, jiraRestClient, userName, password.getPlainText(), readTimeout));
+    }
+
+    private HttpClientOptions getHttpClientOptions() {
         final HttpClientOptions options = new HttpClientOptions();
         options.setRequestTimeout(readTimeout, TimeUnit.SECONDS);
         options.setSocketTimeout(timeout, TimeUnit.SECONDS);
+        options.setCallbackExecutor(getExecutorService());
+        return options;
+    }
 
-        if (executorService==null){
-            int nThreads = threadExecutorNumber;
-            if(nThreads<1){
-                LOGGER.warning( "nThreads " + nThreads + " cannot be lower than 1 so use default " + DEFAULT_THREAD_EXECUTOR_NUMBER );
-                nThreads = DEFAULT_THREAD_EXECUTOR_NUMBER;
-            }
-            executorService =  Executors.newFixedThreadPool(
-                nThreads, //
-                new ThreadFactory()
+    private ExecutorService getExecutorService() {
+        if (executorService==null)
```

Comment:
if only some dependency-injection framework exist in java... :trollface:

Reply:
oh yes not sure how to use it here.... But I will have a look